_sources/UsingRESTAPIs/flickr.rst

```diff
@@ -28,88 +28,71 @@ The structure of a URL for a photo search on flickr is:
 
 Let's put everything together to make a little retrieval tool for flickr images containing particular tags. Of course, in a browser, you can just use flickr's search tool. But doing this through the API opens up other possibilities that you can explore for features not provided on the regular flickr website.
 
-Below is some code that queries the flickr API for images that have a particular tag (I have found that searching for "mountains", "Switzerland", and "cows" usually produces beautiful images that are "safe for work", so the example below does that search.)
-
-.. note:
-
-    To run this code, you will need to copy it to a file on your local machine (not an activecode window), and **paste in an api_key that you get from flickr**.
+Below is some code that queries the flickr API for images that have a particular tag (I have found that searching for "mountains" and "rivers" usually produces beautiful images that are "safe for work", so the example below does that search.)
 
 .. sourcecode:: python
 
+    # import statements
     import requests
     import json
-    import pickle
     import webbrowser
-
-    def canonical_order(d):
```

Comment:
Yes, this is now before caching gets introduced. The book before had flickr code with caching, but introduced the flickr API and caching together. I think it should be changed to first introduce the Flickr API, a more complex API than the previous subchapters have discussed, and then introduce a caching pattern.

I haven't included a function that both uses a caching pattern and gets data from the Flickr API, but that is partially motivated by the fact that doing so is in the problem set I plan to give, once students have written 
* a function that gets data from Flickr
and
* a function that caches data from a simpler API
in earlier psets/class sessions. But I'm not bothered by having it repeated in the textbook and a problem set, if you think it's a better idea.